app/src/main/java/com/github/mateo762/myapplication/upload_gallery/UploadPictureActivity.kt

```diff
@@ -0,0 +1,160 @@
+package com.github.mateo762.myapplication.upload_gallery
+
+import android.app.ProgressDialog
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.github.mateo762.myapplication.databinding.UploadPictureBinding
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.ValueEventListener
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.FirebaseStorage
+import java.text.SimpleDateFormat
+import java.util.*
+
+class UploadPictureActivity : AppCompatActivity() {
+
+    private lateinit var binding: UploadPictureBinding
+    private lateinit var imageUri: Uri
+
+    private var defaultUserString: String = "testUser"
+    private var currentUser: String = defaultUserString
+
+    lateinit var imagesList: ArrayList<UserImage>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = UploadPictureBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.selectPictureButton.setOnClickListener {
+            selectImage()
+        }
+
+        binding.uploadPictureButton.setOnClickListener {
+            uploadImage()
+        }
+
+        binding.imageRecycler.layoutManager = LinearLayoutManager(this)
+        imagesList = arrayListOf()
+
+        //Populate the recycler view with the downloaded images from the current user firebase database
+        var currentUser = Firebase.auth.currentUser?.displayName
+
+        if (currentUser == null) {
+            currentUser = defaultUserString
+        }
+
+        val db = Firebase.database.reference.child("users").child(currentUser)
+        db.addValueEventListener(object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                imagesList = arrayListOf()
+                if (snapshot.exists()) {
+                    for (dataSnapshot in snapshot.children) {
+                        val image = dataSnapshot.getValue(UserImage::class.java)
+                        imagesList.add(image!!)
```

Comment:
!! will crash the app if it is null. And I think that we shouldn't crash if there is no image, we can just drop the image from the list. 

So a better solution would be:
`image?.let {
      imageList.add(it)
}`
Which will add the image if it is not null, otherwise it will not go inside the brackets